addons/mrp/report/mrp_report_mo_overview.py

```diff
@@ -100,7 +100,34 @@ def _get_mo_summary(self, production, components, operations):
         }
 
     def _format_state(self, record):
-        return dict(record._fields['state']._description_selection(self.env)).get(record.state)
+        if record._name != 'mrp.production' or record.state not in ('draft', 'confirmed') or not record.move_raw_ids:
+            return dict(record._fields['state']._description_selection(self.env)).get(record.state)
+        components_qty_to_produce = defaultdict(float)
+        components_qty_reserved = defaultdict(float)
+        for move in record.move_raw_ids:
+            if move.product_id.detailed_type != 'product':
+                continue
+            components_qty_to_produce[move.product_id.id] += move.product_uom._compute_quantity(move.product_uom_qty, move.product_id.uom_id)
+            components_qty_reserved[move.product_id.id] += move.product_uom._compute_quantity(move.reserved_availability, move.product_id.uom_id)
+        producible_qty = record.product_qty
+        for product_id, comp_qty_to_produce in components_qty_to_produce.items():
+            product = self.env['product.product'].browse(product_id)
+            if float_is_zero(comp_qty_to_produce, precision_rounding=product.uom_id.rounding):
+                continue
+            comp_producible_qty = float_round(
+                record.product_qty * (components_qty_reserved[product_id] + product.free_qty) / comp_qty_to_produce,
+                precision_rounding=record.product_uom_id.rounding, rounding_method='DOWN'
+            )
+            if float_is_zero(comp_producible_qty, precision_rounding=record.product_uom_id.rounding):
+                producible_qty = 0
```

Comment:
Why not return `not ready` here?
`record.product_qty` can't be zero, I think it's ok just return it here and no need to check it again out of the loop